src/Storages/HDFS/StorageHDFS.cpp

```diff
@@ -330,6 +323,22 @@ BlockOutputStreamPtr StorageHDFS::write(const ASTPtr & /*query*/, const StorageM
         chooseCompressionMethod(uri, compression_method));
 }
 
+void StorageHDFS::truncate(const ASTPtr & /* query */, const StorageMetadataPtr &, ContextPtr context_, TableExclusiveLockHolder &)
+{
+    const size_t begin_of_path = uri.find('/', uri.find("//") + 2);
+    const String path = uri.substr(begin_of_path);
+    const String url = uri.substr(0, begin_of_path);
+
+    HDFSBuilderWrapper builder = createHDFSBuilder(url + "/", context_->getGlobalContext()->getConfigRef());
+    HDFSFSPtr fs = createHDFSFS(builder.get());
+
+    int wait;
+    int ret = hdfsTruncate(fs.get(), path.data(), 0, &wait);
```

Comment:
wait result is unused? what are the semantics?

Reply:
It is different now.